src/liballoc/boxed.rs

```diff
@@ -763,6 +783,21 @@ impl<A, F> FnBox<A> for F
     }
 }
 
+#[cfg(not(stage0))]
+#[unstable(feature = "fnbox",
+           reason = "will be deprecated if and when `Box<FnOnce>` becomes usable", issue = "28796")]
+impl<A, F> FnBox<A> for F
+    where F: FnOnce<A>,
+          <F as FnOnce<A>>::Output: marker::Move,
```

Comment:
Why do we have this requirement? Isn't this required by `FnOnce` for back-compat reasons?